flink-yarn/src/main/java/org/apache/flink/yarn/FlinkYarnClient.java

```diff
@@ -341,7 +342,7 @@ public AbstractFlinkYarnCluster run() throws Exception {
 	 * This method will block until the ApplicationMaster/JobManager have been
 	 * deployed on YARN.
 	 */
-	protected AbstractFlinkYarnCluster deployInternal(String clusterName) throws Exception {
+	protected AbstractFlinkYarnCluster deployInternal(String defaultName) throws Exception {
```

Comment:
Since this becomes default, maybe we could just remove the input argument and set default internally if customName is null or empty.